tests/scripts/all.sh

```diff
@@ -185,7 +185,7 @@ pre_initialize_variables () {
     export CTEST_OUTPUT_ON_FAILURE=1
 
     # CFLAGS and LDFLAGS for Asan builds that don't use CMake
-    ASAN_CFLAGS='-Werror -Wall -Wextra -fsanitize=address,undefined -fno-sanitize-recover=all'
+    ASAN_CFLAGS='-O2 -Werror -fsanitize=address,undefined -fno-sanitize-recover=all'
```

Comment:
We aren't passing `-Wall -Wextra` in `programs/fuzz`… That has a broader impact, so this isn't the place to work around it. I'll make a PR to fix `programs/fuzz/Makefile`.

Reply:
https://github.com/Mbed-TLS/mbedtls/pull/6687